src/main/java/uk/gov/pay/directdebit/mandate/params/MandateSearchParams.java

```diff
@@ -101,18 +107,47 @@ public MandateSearchParams withToDate(ZonedDateTime toDate) {
         return this;
     }
 
-    public MandateSearchParams withPage(int page) {
-        this.page = page;
+    public MandateSearchParams withPage(Integer page) {
+        if (page != null && page < 1) {
+            throw new BadRequestException("Query param 'page' should be a non zero positive integer.");
+        }
+        this.page = Optional.ofNullable(page).orElse(0);
         return this;
     }
 
-    public MandateSearchParams withDisplaySize(int displaySize) {
-        this.displaySize = displaySize;
+    public MandateSearchParams withDisplaySize(Integer displaySize) {
+        int size = Optional.ofNullable(displaySize).orElse(500);
+        if (!Range.closed(1, 500).contains(size)) {
```

Comment:
It seems that when searching for Direct Debit payments we don’t actually check if the value is greater than 500, which is surprising, but we probably want to make it consistent one way or the other